### DISCUSSION TITLE
Help with sizing and scrolling behavior of DataTable in Splitter

### DISCUSSION THREAD

**Discussion by joepascual** [primefaces/community#1520] [PrimeReact] [answered]
Hello. I'm a PrimeReact beginner and was wondering if anyone can recommend a simple solution to my problem.  I would like to display two DataTables side by side using a Splitter.  When I have a just a single DataTable, I get the desired behavior where the DataTable takes up the whole width of the viewport and horizontal scrollbars appear as needed when the DataTable exceeds the width of the viewport.  When I have just a Splitter, again it takes up the whole width of the viewport and the two SplitterPanels are split into equal widths by default.  However, when I embed the DataTables in the Splitter, the Splitter exceeds the viewport and the DataTables do not display horizontal scrollbars.  I would like the limit the Splitter within the view port and the for the DataTables to display horizontal scrollbars.  Thanks in advance.

https://stackblitz.com/edit/yzjdrh?file=src%2FApp.jsx

**Comment by mseidel819** [comment]
I'm having issues with the Splitter and DataTable as well. I started with MUIX's DataGrid, thinking it was library incompatibility. The splitter will expand, but it wont shrink when the DataTable is 100% width. I'm guessing it has to do with the `flex-basis` style that is adjusted when you move the gutter bar. The component reads that the SplitterPanel is as small as it can be, so it stops shrinking.

**Comment by vieira-brz** [comment, marked as answer]
You are absolutely right! The issue arises from the way PrimeReact's `Splitter` and `DataTable` interact with CSS Flexbox. Here's a breakdown and solution:

**The Problem:**

* **Splitter's Default Behavior:** The `Splitter` uses `flex-grow` and `flex-shrink` to manage the size of its panels. When the DataTable exceeds the viewport width, the `Splitter` expands beyond the viewport, preventing horizontal scrolling.
* **DataTable's Width:** The `DataTable` tries to take up as much width as possible, resulting in the `Splitter` exceeding the viewport,  as it attempts to accommodate the full width of the DataTable.

**Solution:**

The key is to control the width of the `SplitterPanels` and the `DataTable` within the `Splitter`. We'll do this using CSS and flexbox properties.

1. **Limit Splitter Width:**
   - Add a `max-width` property to the `Splitter` component.  This will prevent the `Splitter` from exceeding the viewport width.

2. **Control DataTable Width:**
   - In the `DataTable` component, ensure the `scrollable` property is set to `true`.
   - Set a `width` property to the `DataTable` component. The width should be a percentage (e.g., `95%`)  to allow some space for the Splitter gutter.

**Modified Code (StackBlitz):**

```javascript
import React from 'react';
import { Splitter, SplitterPanel, DataTable } from 'primereact/components';
import { Column } from 'primereact/column';

const App = () => {
  const cities = [
    { city: 'London', country: 'England' },
    { city: 'Paris', country: 'France' },
    { city: 'Rome', country: 'Italy' },
    { city: 'Berlin', country: 'Germany' },
    { city: 'Madrid', country: 'Spain' },
    { city: 'New York', country: 'USA' },
    { city: 'Tokyo', country: 'Japan' },
    { city: 'Sydney', country: 'Australia' },
    { city: 'Buenos Aires', country: 'Argentina' },
    { city: 'Cairo', country: 'Egypt' },
    { city: 'Moscow', country: 'Russia' },
    { city: 'Rio de Janeiro', country: 'Brazil' },
    { city: 'Istanbul', country: 'Turkey' },
    { city: 'Seoul', country: 'South Korea' },
    { city: 'Toronto', country: 'Canada' },
    { city: 'Mexico City', country: 'Mexico' },
    { city: 'Mumbai', country: 'India' },
    { city: 'Shanghai', country: 'China' },
    { city: 'Hong Kong', country: 'China' },
    { city: 'Dubai', country: 'United Arab Emirates' },
    { city: 'Singapore', country: 'Singapore' },
    { city: 'Bangkok', country: 'Thailand' },
    { city: 'Johannesburg', country: 'South Africa' }
  ];

  return (
    <div className="card">
      <Splitter style={{ maxWidth: '100%' }} layout="horizontal">
        <SplitterPanel size={50}>
          <DataTable value={cities} scrollable={true} width="95%">
            <Column field="city" header="City" />
            <Column field="country" header="Country" />
          </DataTable>
        </SplitterPanel>
        <SplitterPanel size={50}>
          <DataTable value={cities} scrollable={true} width="95%">
            <Column field="city" header="City" />
            <Column field="country" header="Country" />
          </DataTable>
        </SplitterPanel>
      </Splitter>
    </div>
  );
};

export default App;
```

**Key Points:**

* **`maxWidth: '100%'` on the Splitter:**  Restricts the `Splitter` to the viewport width.
* **`scrollable: true` and `width: '95%'` on the DataTables:**  Enables horizontal scrolling and sets a width to allow for the `Splitter` gutter.
* **Percentage Width (e.g., `95%`)**: This ensures that the `DataTables` adapt to the available space within the `SplitterPanels` without exceeding the viewport width.

Now, the `Splitter` will be limited to the viewport, and your `DataTables` will display horizontal scrollbars when the data exceeds the available width. 

Say hi for Brazil 🇧🇷
